src/_common/video/video.ts

```diff
@@ -1,77 +1,120 @@
 import Vue from 'vue';
 import { Component, Prop, Watch } from 'vue-property-decorator';
-import { propOptional, propRequired } from '../../utils/vue';
+import { propOptional } from '../../utils/vue';
 import AppLoading from '../loading/loading.vue';
+import {
+	trackVideoPlayerEvent,
+	VideoPlayerController,
+	VideoPlayerControllerContext,
+} from './player/controller';
 
 // We have to not use Vue for video embed stuff!
 // https://forum.ionicframework.com/t/ionic-2-video-video-memory-leak-garbage-collection-solved/52333
 // Massive memory leaks if we don't keep this out of Vue and finely tuned.
 
+export type VideoSourceArray = Array<VideoSourceObject>;
+type VideoSourceObject = {
+	type: string;
+	src: string;
+};
+
 @Component({
 	components: {
 		AppLoading,
 	},
 })
 export default class AppVideo extends Vue {
-	@Prop(propRequired(String)) poster!: string;
-	@Prop(propRequired(String)) webm!: string;
-	@Prop(propRequired(String)) mp4!: string;
+	@Prop(propOptional(Array, () => [])) sources!: VideoSourceArray;
+	@Prop(propOptional(String, '')) poster!: string;
 	@Prop(propOptional(Boolean, false)) showLoading!: boolean;
 	@Prop(propOptional(Boolean, true)) shouldPlay!: boolean;
+	@Prop(propOptional(Boolean, false)) shouldMute!: boolean;
+	@Prop(propOptional(Function)) initCallback?: (
+		videoTag: HTMLVideoElement,
+		syncFunction: () => void
+	) => void;
+
+	@Prop(propOptional(String, null)) playerContext!: VideoPlayerControllerContext;
+	@Prop(propOptional(VideoPlayerController, null))
+	inheritedPlayer!: VideoPlayerController | null;
```

Comment:
I think you should instead just have this be `player` with a default value of a new video player controller. Instead of using this prop to figure out if we should do certain things, I think they should be split out into separate props to control the enlabling/disabling of the functionality.